### PR TITLE
Update Frontier machine/compilers following system update

### PR DESCRIPTION
**Pull request by grnydawn**
- Retain current software modules instead of updating to the latest versions to prioritize reliability.
- Add linker options to use GCC 12.2, addressing linker errors.
- Utilize Fortran linker to resolve additional linker errors.
- Replace hipcc with mpicxx for MPICXX macro in the GPU compiler definitions.
- Adjust compiler priority to prioritize reliability over performance.
- Temporarily comment out ADIOS2 configurations

### CODE REVIEW

**Comment by rljacob** [comment]
How was this tested?

**Comment by grnydawn** [comment]
The following table summarizes the test results. Some results are from running the e3sm_developer test suite, while others are from running individual test cases in the e3sm_developer test suite without debug cases. It took an excessive amount of time to build debug cases with the crayclang compiler.

| Test Result       | crayclang (Current) | crayclang (Latest) | amdclang (Current) | amdclang (Latest) | gnu (Current) | gnu (Latest) |
|-------------------|---------------------|--------------------|--------------------|-------------------|---------------|--------------|
| PASS              | 63                  | 71                 | 58                 | 82                | 65            | 71           |
| FAIL              | 9                   | 11                 | 14                 | 44                | 7             | 12           |
| Total             | 72                  | 82                 | 72                 | 126               | 72            | 83           |
| Avg. Build Time (secs)   | 1137                | 1619               | 289                | 293               | 133           | 123          |

**Comment by grnydawn** [comment]
The crayclang compiler (both the current and latest versions) has internal compiler issues and excessive compile times (OLCF tickets for the latest version: OLCFHELP-19210, OLCFHELP-19356, and OLCFHELP-19435).

The amdclang compiler has a segfault issue during the compilation of some test cases. Essentially, AMD has asked us to wait until they release a new Fortran compiler based on LLVM.

**Review by jgfouca — APPROVED** [review]

**Comment by rljacob** [comment]
Try testing with this suite:  e3sm_gpucxx

**Comment by grnydawn** [comment]
@rljacob , I got following test result with e3sm_gpucxx test suite.

|              | amdclang           | amdclanggpu       | crayclang           | crayclanggpu       | gnu               | gnugpu            |
|--------------|--------------------|-------------------|---------------------|--------------------|-------------------|-------------------|
| **SCREAM**   | FAIL MODEL_BUILD    | FAIL SHAREDLIB_BUILD | FAIL MODEL_BUILD    | FAIL MODEL_BUILD   | FAIL MODEL_BUILD  | FAIL MODEL_BUILD  |
| **MMF**      | PASS                | PASS              | PEND MODEL_BUILD    | PEND MODEL_BUILD   | PASS              | PASS              |

All scream failures except amdclanggpu compiler case are caused by the following error:

```
-- Configuring incomplete, errors occurred!
...
CMake Error at cmake/build_eamxx.cmake:34 (include):
include could not find requested file:
No macro file found: .../cmake_macros/frontier.cmake in build_eamxx.cmake
```

**Comment by grnydawn** [comment]
The following cmake statements in "$E3SM/components/cmake/build_eamxx.cmake" caused the error:
```
set(SCREAM_MACH_FILE_ROOT ${CMAKE_SOURCE_DIR}/eamxx/cmake/machine-files)
if (EXISTS ${SCREAM_MACH_FILE_ROOT}/${MACH}-${COMPILER}.cmake)
    include(${SCREAM_MACH_FILE_ROOT}/${MACH}-${COMPILER}.cmake)
else()
    include(${SCREAM_MACH_FILE_ROOT}/${MACH}.cmake)
```

The only cmake file that Scream currently supports for Frontier is "frontier-scream-gpu.cmake". So, none of the above machine/compiler names worked.

**Comment by jgfouca** [comment]
@grnydawn , eamxx won't work if there's no eamxx machine file. Eamxx clears all E3SM settings and manages its own flags, so you will end up with no compiler flags if there's no eamxx machine file.

**Comment by rljacob** [comment]
What stops EAMxx from using the regular E3SM build system components?

**Comment by jgfouca** [comment]
@rljacob , a long time ago, we thought eamxx would need to manage it's own flags, even when part of a CIME build. I think we have since decided that we should just use the CIME system but I haven't gotten around to changing things yet.

**Review by rljacob — APPROVED** [review]

**Comment by grnydawn** [comment]
If there are no objections, I will merge this PR to the next branch, and, if no new issues arise, into the master branch.

**Pull request merged**
